# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: 5dfb9fc15fba
# Full Hash: 5dfb9fc15fba710d15d5720e03f6bb398d7b6fbd
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 16:09:44
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 4: Untie frame iteration from Frame::tls. r=lth
#   
#   
#   Here we replace usage of Frame::tls in frame iteration with GetNearestEffectiveTls.
#   We also maintain current tls for frame iteration object to not to call GetNearestEffectiveTls everytime.
# ==============================================================================

diff -r 3cfa16c9f9a9 -r 5dfb9fc15fba js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Fri Oct 23 18:17:37 2020 +0000
+++ b/js/src/jit/MacroAssembler.h	Fri Oct 23 18:17:37 2020 +0000
@@ -4411,10 +4411,13 @@
       : ABIArgIterBase<VecT, WasmABIArgGenerator>(types) {}
 };
 
-wasm::TlsData* ExtractCalleeTlsFromFrameWithTls(wasm::Frame* fp);
-wasm::TlsData* ExtractCallerTlsFromFrameWithTls(wasm::Frame* fp);
-
 }  // namespace jit
+
+namespace wasm {
+TlsData* ExtractCalleeTlsFromFrameWithTls(Frame* fp);
+TlsData* ExtractCallerTlsFromFrameWithTls(Frame* fp);
+}  // namespace wasm
+
 }  // namespace js
 
 #endif /* jit_MacroAssembler_h */